### PR TITLE
feat:added fontawesome + some elements

### PR DESCRIPTION
**Pull request by kriskarpenko**
Feat: added FontAwesome: contact button, menu bars. 
Fix:  contact button, Section

### CODE REVIEW

File: components/Contact/Contact.js [hunk 1]
```diff
@@ -1,10 +1,19 @@
...
 
 const Contact = () => {
   return (
-    <div>
-      <button className={styles.contactButton}>Contact</button>
+    <div className={styles.contactButton}>
+      <Link href="./contact">
+        <button>
+          <div>
+            <FontAwesomeIcon icon={faPhoneVolume} width="30" />
```
Comment (karpolan) [inline components/Contact/Contact.js:13]: 16, 24, 32, 48 are standard size of icons.

But you can use any...

File: components/Contact/Contact.js [hunk 2]
```diff
@@ -1,10 +1,19 @@
...
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faPhoneVolume } from "@fortawesome/free-solid-svg-icons";
+import Link from "next/link";
 
 const Contact = () => {
   return (
-    <div>
-      <button className={styles.contactButton}>Contact</button>
+    <div className={styles.contactButton}>
+      <Link href="./contact">
```
Comment (karpolan) [inline components/Contact/Contact.js:10]: I think it should be "/contact"

For links:

/name - in the root of website
../name - get one level app
name - same level (as "./name" in files)

File: components/Contact/Contact.module.css [hunk 3]
```diff
@@ -2,5 +2,6 @@
   display: block;
   padding: 1%;
```
Comment (karpolan) [inline components/Contact/Contact.module.css:3]: Try to resize font size (Cmd + / -)

padding, margins, borders in precents - are bad idea

File: components/Contact/Contact.module.css [hunk 4]
```diff
@@ -2,5 +2,6 @@
...
   margin: 2%;
-  margin-left: auto;
+  display: block;
+  shape-rendering: geometricPrecision;
```
Comment (karpolan) [inline components/Contact/Contact.module.css:6]: Wow, what it does?

File: components/NavBar/NavComponents/NavBarMobile.js [hunk 5]
```diff
@@ -17,7 +20,13 @@ const NavBarMobile = () => {
 
   return (
     <div className={styles.mobile}>
-      <button onClick={onMenuTogle}>{menuOpen ? "X" : "?"}</button>
+      <button onClick={onMenuTogle}>
+        {menuOpen ? (
+          <FontAwesomeIcon icon={faXmark} width="16" />
+        ) : (
+          <FontAwesomeIcon icon={faBars} width="20" />
```
Comment (karpolan) [inline components/NavBar/NavComponents/NavBarMobile.js:27]: Why icon sizes are different?

File: components/Section/Section3.js [hunk 6]
```diff
@@ -2,7 +2,7 @@ import styles from "./Section.module.css";
 
 const Section = ({ children, pictureAfterText = false, picture, text }) => {
   return (
-    <p className={pictureAfterText ? styles.sectionRight : styles.sectionLeft}>
+    <p className={pictureAfterText ? styles.sectionLeft : styles.sectionRight}>
```
Comment (karpolan) [inline components/Section/Section3.js:5]: I think for mobile it should one styling with picture before text

File: package.json [hunk 7]
```diff
@@ -9,6 +9,11 @@
     "lint": "next lint"
   },
   "dependencies": {
+    "@fortawesome/fontawesome-svg-core": "^6.2.0",
+    "@fortawesome/free-regular-svg-icons": "^6.2.0",
+    "@fortawesome/free-solid-svg-icons": "^6.2.0",
+    "@fortawesome/react-fontawesome": "^0.2.0",
+    "babel-plugin-macros": "^3.1.0",
```
Comment (karpolan) [inline package.json:16]: Is it really needed?

File: package.json [hunk 8]
```diff
@@ -9,6 +9,11 @@
     "lint": "next lint"
   },
   "dependencies": {
+    "@fortawesome/fontawesome-svg-core": "^6.2.0",
+    "@fortawesome/free-regular-svg-icons": "^6.2.0",
```
Comment (karpolan) [inline package.json:13]: BTW, it is possible to add CDN link to FontAwesone in <head> and use html tags to render icons.

React components is also fine.

File: pages/_app.js [hunk 9]
```diff
@@ -4,8 +4,13 @@ import Head from "next/head";
 import { CONFIG } from "../utils/consts";
 import "../styles/globals.css";
 import Contact from "../components/Contact/Contact";
+import { useRouter } from "next/router";
 
 function MyApp({ Component, pageProps }) {
+  const router = useRouter();
+  const linkContact = () => {
+    router.pathname === "./contact" ? null : <Contact />;
```
Comment (karpolan) [inline pages/_app.js:12]: Why you need this?

Contact is need on `/page/contact.js` only

File: pages/_app.js [hunk 10]
```diff
@@ -4,8 +4,13 @@ import Head from "next/head";
 import { CONFIG } from "../utils/consts";
 import "../styles/globals.css";
 import Contact from "../components/Contact/Contact";
+import { useRouter } from "next/router";
 
 function MyApp({ Component, pageProps }) {
+  const router = useRouter();
```
Comment (karpolan) [inline pages/_app.js:10]: Bad idea...

File: pages/_app.js [hunk 11]
```diff
@@ -16,7 +21,8 @@ function MyApp({ Component, pageProps }) {
 
       <NavBar />
       <Component {...pageProps} />
-      <Contact />
+      {linkContact}
```
Comment (karpolan) [inline pages/_app.js:24]: Contact is part of layout? 

It should be on separate page. 

Or move it to the Footer

File: pages/contentMain/ContentMain.js [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+const ContentMain = () => {
+  return (
+    <div className={styles.gridContainer}>
+      <p className={styles.titleLeft}>We are</p>
+      <div className={styles.description}>
+        <Section
+          text="A new generation singing school. A place where you can fullfil your
+        dream, of becoming greate singer, with only one year of learning."
+          picture={
+            <img
```
Comment (karpolan) [inline pages/contentMain/ContentMain.js:14]: Better use Image component from Next instead of img tag

File: pages/contentMain/ContentMain.js [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+              // className={styles.onOverlay}
+              src="/logo.jpeg"
```
Comment (karpolan) [inline pages/contentMain/ContentMain.js:16]: Bad naming, bad location

/public/img/moto.jpg 

or 

/public/assets/section/moto.jpg


Also it is possible to use images in JS project, import them as .css

File: pages/contentMain/ContentMain.js [hunk 14]
```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import Section from "../../components/Section";
+import styles from "./ContentMain.module.css";
+
+const ContentMain = () => {
+  return (
+    <div className={styles.gridContainer}>
+      <p className={styles.titleLeft}>We are</p>
+      <div className={styles.description}>
```
Comment (karpolan) [inline pages/contentMain/ContentMain.js:9]: Why you need div? 

Make Section to support className property

You can combine classes as strings or use `clsx()` fucntion from `clsx` npm

File: pages/contentMain/ContentMain.js [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+            <img
+              // className={styles.onOverlay}
+              src="/logo.jpeg"
+              width="400"
+              height="500"
+            />
+          }
+        ></Section>
+      </div>
+      <div className={styles.gridContainerDescribtion}>
```
Comment (karpolan) [inline pages/contentMain/ContentMain.js:23]: You don't need additional div here

File: pages/contentMain/ContentMain.module.css [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+  font-family: serif;
+  margin: 0;
+  line-height: 1.15;
+  font-size: 4rem;
+  text-transform: uppercase;
+}
+
+.gridContainer {
+  display: grid;
+  grid-auto-rows: auto;
```
Comment (karpolan) [inline pages/contentMain/ContentMain.module.css:20]: Why?

File: pages/contentMain/ContentMain.module.css [hunk 17]
```diff
@@ -0,0 +1,52 @@
+.contentMobile {
+  max-width: 720px;
+  display: flex;
+  gap: 1rem;
+  align-items: center;
+  overflow: visible;
+  grid: none;
```
Comment (karpolan) [inline pages/contentMain/ContentMain.module.css:7]: remove

File: pages/contentMain/ContentMain.module.css [hunk 18]
```diff
@@ -0,0 +1,52 @@
+.contentMobile {
```
Comment (karpolan) [inline pages/contentMain/ContentMain.module.css:1]: Where is .contentDesktop ?

Why you actually need this?

File: pages/contentMain/ContentMain.module.css [hunk 19]
```diff
@@ -0,0 +1,52 @@
...
+  text-transform: uppercase;
+}
+
+.gridContainer {
+  display: grid;
+  grid-auto-rows: auto;
+}
+
+.titleLeft {
+  font-family: serif;
```
Comment (karpolan) [inline pages/contentMain/ContentMain.module.css:24]: bad

File: pages/contentMain/ContentMain.module.css [hunk 20]
```diff
@@ -0,0 +1,52 @@
...
+  line-height: 1.15;
+  font-size: 4rem;
```
Comment (karpolan) [inline pages/contentMain/ContentMain.module.css:26]: bad

File: pages/contentMain/ContentMain.module.css [hunk 21]
```diff
@@ -0,0 +1,52 @@
...
+  margin-left: 15%;
```
Comment (karpolan) [inline pages/contentMain/ContentMain.module.css:27]: WTF?

File: pages/contentMain/ContentMain.module.css [hunk 22]
```diff
@@ -0,0 +1,52 @@
...
+  text-transform: uppercase;
+}
+
+.titleRight {
+  font-family: serif;
```
Comment (karpolan) [inline pages/contentMain/ContentMain.module.css:32]: bad

File: pages/contentMain/ContentMain.module.css [hunk 23]
```diff
@@ -0,0 +1,52 @@
...
+  line-height: 1.15;
+  font-size: 4rem;
+  margin-left: auto;
```
Comment (karpolan) [inline pages/contentMain/ContentMain.module.css:35]: It is hacking

File: pages/contentMain/ContentMain.module.css [hunk 24]
```diff
@@ -0,0 +1,52 @@
...
+.title,
+.description {
+  text-align: center;
+}
+
+.description {
+  margin: 0 0;
+  line-height: 1.5;
+  font-size: 1.5rem;
+  font-family: serif;
```
Comment (karpolan) [inline pages/contentMain/ContentMain.module.css:48]: bad

File: pages/contentMain/ContentMain.module.css [hunk 25]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+.title,
+.description {
+  text-align: center;
+}
+
+.description {
+  margin: 0 0;
+  line-height: 1.5;
```
Comment (karpolan) [inline pages/contentMain/ContentMain.module.css:46]: Why do you set .line-height everywhere?

File: pages/contentMain/ContentMain.module.css [hunk 26]
```diff
@@ -0,0 +1,52 @@
...
+  font-size: 1.5rem;
+  font-family: serif;
+  color: #464646;
+  justify-self: center;
```
Comment (karpolan) [inline pages/contentMain/ContentMain.module.css:50]: bad

File: pages/contentMain/ContentMain.module.css [hunk 27]
```diff
@@ -0,0 +1,52 @@
...
+  width: 80%;
```
Comment (karpolan) [inline pages/contentMain/ContentMain.module.css:51]: very bad

File: pages/contentMain/MainMobile.js [hunk 28]
```diff
@@ -0,0 +1,12 @@
+import ContentMain from "./ContentMain";
```
Comment (karpolan) [inline pages/contentMain/MainMobile.js:1]: Never put components in page folder!

Try to open `http://loacalhost:300/contentMain/MainMobile` link :)

File: pages/index.js [hunk 29]
```diff
@@ -1,30 +1,10 @@
...
-        </div>
-        <div className={styles.gridContainerDescribtion}>
-          <Section></Section>A new generation singing school.
-        </div>
-      </div>
-    </main>
-  );
+  const isMobile = useMediaQuery({ query: "(max-width: 720px)" });
+
+  return <div>{isMobile ? <MainMobile /> : <ContentMain />}</div>;
```
Comment (karpolan) [inline pages/index.js:9]: Common! 

Will you add same code to every page?

Move it to Layout, it is `app.js` for now

File: styles/Home.module.css [hunk 30]
```diff
@@ -53,12 +53,23 @@
   text-transform: uppercase;
 }
 
+.gridContainer {
+  display: grid;
+}
+
 .titleLeft {
```
Comment (karpolan) [inline styles/Home.module.css:60]: Same class in different .css

File: styles/Home.module.css [hunk 31]
```diff
@@ -53,12 +53,23 @@
   text-transform: uppercase;
 }
 
+.gridContainer {
```
Comment (karpolan) [inline styles/Home.module.css:56]: Why?

File: styles/globals.css [hunk 32]
```diff
@@ -11,19 +11,67 @@ a {
...
+}
 * {
   box-sizing: border-box;
 }
 
+.gridContainer {
+  display: grid;
+}
+
+.titleLeft {
```
Comment (karpolan) [inline styles/globals.css:35]: Same again

File: styles/globals.css [hunk 33]
```diff
@@ -11,19 +11,67 @@ a {
...
 }
 
+div {
+  margin: 0 0;
+}
 * {
   box-sizing: border-box;
 }
 
+.gridContainer {
```
Comment (karpolan) [inline styles/globals.css:31]: Same class again

File: styles/globals.css [hunk 34]
```diff
@@ -11,19 +11,67 @@ a {
...
 p {
   margin-block-end: 0rem;
   margin-block-start: 0rem;
+  color: #2e2e2e;
 }
 
+div {
+  margin: 0 0;
+}
 * {
```
Comment (karpolan) [inline styles/globals.css:27]: WTF?

File: styles/globals.css [hunk 35]
```diff
@@ -11,19 +11,67 @@ a {
...
+  margin: 0 0;
+  line-height: 1.5;
+  font-size: 1.5rem;
+  font-family: serif;
+  color: #464646;
+  justify-self: center;
+  width: 80%;
+}
+
+button {
```
Comment (karpolan) [inline styles/globals.css:66]: Global styling? Really?

File: styles/globals.css [hunk 36]
```diff
@@ -11,19 +11,67 @@ a {
...
+  margin-left: auto;
+  text-transform: uppercase;
+}
+
+.title,
+.description {
+  text-align: center;
+}
+
+.description {
```
Comment (karpolan) [inline styles/globals.css:56]: Same class again

File: utils/ babel-plugin-macros.config.js [hunk 37]
```diff
@@ -0,0 +1,7 @@
+import {} from "@fortawesome/fontawesome-svg-core/import.macro";
```
Comment (karpolan) [inline utils/ babel-plugin-macros.config.js:1]: So FontAwesome was not working without this shit?

File: utils/babel.config.js [hunk 38]
```diff
@@ -0,0 +1,5 @@
+module.exports = function (api) {
```
Comment (karpolan) [inline utils/babel.config.js:1]: Hacking...

File: utils/ babel-plugin-macros.config.js (hunk 37 above)
Comment (kristinakarpenko) [reply utils/ babel-plugin-macros.config.js:1]: mhm there are bunch of installs and imports and also some bubble files 🙃

File: components/Contact/Contact.js [hunk 39]
```diff
@@ -1,10 +1,19 @@
...
+        <button>
+          <div>
```
[the remark this answers is not in the document]
Comment (kristinakarpenko) [reply components/Contact/Contact.js:12]: u mean <Tab/>?

[the remark this answers is not in the document]
Comment (karpolan) [reply components/Contact/Contact.js:12]:
```
 <Link href="/contact">
   <FontAwesomeIcon icon={faPhoneVolume} width="30" />
</Link>
```

File: utils/ babel-plugin-macros.config.js (hunk 37 above)
Comment (karpolan) [reply utils/ babel-plugin-macros.config.js:1]: I suggest to use some React Styling Library, or FontAwesome as CDN 

```
<head>
<link rel="stylesheet" href="path-or-CDN-to/font-awesome/css/font-awesome.min.css">
</head>

...
<i class="fa-solid fa-user"></i>
OR
<span class="fa-solid fa-user"></span>
```

**Pull request closed without merging**